ffi/wgpu.h

```diff
@@ -980,7 +980,8 @@ void wgpu_sampler_destroy(WGPUSamplerId sampler_id);
 #endif
 
 #if defined(WGPU_LOCAL)
-WGPUSwapChainOutput wgpu_swap_chain_get_next_texture(WGPUSwapChainId swap_chain_id);
+int32_t wgpu_swap_chain_get_next_texture(WGPUSwapChainOutput *swap_chain_output,
```

Comment:
Let's try to keep the header unchanged, or otherwise it would be more difficult to get #47 later.
Perhaps, the `WGPUSwapChainOutput` could have a special value indicating an error?

Reply:
Good consideration. It seems fairly important to be able to easily identify errors here though - there's no leeway for changing the header at all?

The only way to do this without changing the header would be to designate special values for the internal `uint64_t view_id`. But special values should probably also conform to a spec rather than just being arbitrarily selected...

Reply:
We don't have a clear picture on how the error handling will work on the "webgpu.h" yet. We are likely going to have *some* mechanism of querying the error states, like the Web side. So for the functions that may fail, like this one, we don't need to return `Result` semantics, we basically need `Option`. So having a special ID value for the error (like "NULL") is fine. More to that, in Dawn it *is* actually a pointer, so NULL makes sense. For us, we don't expect any ID value to be 0 (at least, for the swapchain stuff), so it should be fine to return it from here.